Mathlib/RingTheory/Ideal/IsPrincipalPowQuotient.lean

```diff
@@ -0,0 +1,86 @@
+/-
+Copyright (c) 2024 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import Mathlib.LinearAlgebra.Isomorphisms
+import Mathlib.RingTheory.Ideal.Operations
+import Mathlib.RingTheory.Ideal.Quotient
+
+/-!
+# Quotients of powers of principal ideals
+
+This file deals with taking quotients of powers of principal ideals.
+
+## Main definitions and results
+
+* `Ideal.quot_equiv_pow_quot_pow_succ`: for a principal ideal `I`, `R ⧸ I ≃ₗ[R] I ^ n ⧸ I ^ (n + 1)`
+
+## Implementation details
+
+At site of usage, calling `LinearEquiv.toEquiv` can cause timeouts in the search for a complex
+synthesis like `Module 𝒪[K] 𝓀[k]`, so the plain equiv versions are provided.
+
+These equivs are defined here as opposed to in the quotients file since they cannot be
+formed as ring equivs.
+
+-/
+
+
+namespace Ideal
+
+section IsPrincipal
+
+variable {R : Type*} [CommRing R] [IsDomain R] {I : Ideal R}
+
+/-- For a principal ideal `I`, `R ⧸ I ≃ₗ[R] I ^ n ⧸ I ^ (n + 1)`. To convert into a form
+that uses the ideal of `R ⧸ I ^ (n + 1)`, compose with
+`Ideal.pow_quot_pow_succ_equiv_map_mk_pow_succ_pow`. -/
+noncomputable
+def quot_equiv_pow_quot_pow_succ (h : I.IsPrincipal) (h': I ≠ ⊥) (n : ℕ) :
```

Comment:
Hmm, actually, shouldn't this use camelCase?

Reply:
changed